src/Auth.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useEffect } from "react";
+import {
+    useLocation,
+    Navigate,
+} from "react-router-dom";
+import { Role, User } from "./models/Users";
+import { login } from "./logic/api";
+
+interface AuthContextType {
+    user: User | null; 
+    token: string | null;
+    role: Role | null;
+
+    signIn: (role: Role, email: string, password: string) => void;
+    signOut: () => void;
+}
```

Comment:
Nie wiem czy widziałeś, AuthContextType przeniosłem do /components/types.tsx, wartoby było to uwzględnić